llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -3606,6 +3613,32 @@ bool SPIRVInstructionSelector::selectSpvThreadId(Register ResVReg,
   return Result && MIB.constrainAllUses(TII, TRI, RBI);
 }
 
+bool SPIRVInstructionSelector::selectSpvThreadId(Register ResVReg,
+                                                 const SPIRVType *ResType,
+                                                 MachineInstr &I) const {
+  // DX intrinsic: @llvm.dx.thread.id(i32)
+  // ID  Name      Description
+  // 93  ThreadId  reads the thread ID
+  //
+  // In SPIR-V, llvm.dx.thread.id maps to a `GlobalInvocationId` builtin
+  // variable
+  return loadVec3BuiltinInputID(SPIRV::BuiltIn::GlobalInvocationId, ResVReg,
+                                ResType, I);
```

Comment:
I'm not sure I see any advantage of wrapping these calls in these functions rather than calling `loadVec3BuiltinInputID` directly from the switch case in `selectIntrinsic`.

Reply:
@tex3d Yeah, I noticed this as well. I don't call `loadVec3BuiltinInputID` directly just for keep the change minimal.

Reply:
I don't see why we would need to change it in the future, and in any case, adding a function to do something more complicated can always be done in the future if necessary.

Reply:
@tex3d  Done. Updated the PR based on your comments.